### PR TITLE
[#4300] black/white microservice authentication add apiWhiteList

### PR DESCRIPTION
**Pull request by chengyouling**

### CODE REVIEW

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java [hunk 1]
```diff
@@ -50,11 +57,44 @@ public String getName() {
 
   @Override
   public CompletableFuture<Response> onFilter(Invocation invocation, FilterNode nextNode) {
+    if (checkPathWhitelist(invocation.getRequestEx().getServletPath())) {
```
Comment (liubao68) [inline handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java:60]: Do you mean request url by using `invocation.getRequestEx().getServletPath()` ?

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java [hunk 2]
```diff
@@ -27,15 +28,21 @@
 import org.apache.servicecomb.swagger.invocation.Response;
 import org.apache.servicecomb.swagger.invocation.exception.InvocationException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
 
 import jakarta.ws.rs.core.Response.Status;
 
 public class ProviderAuthFilter extends AbstractFilter implements ProviderFilter {
+  private static final String KEY_API_WHITE_LIST = "servicecomb.publicKey.accessControl.pathWhiteList";
```
Comment (liubao68) [inline handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java:36]: servicecomb.publicKey.accessControl.exclude

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java (hunk 1 above)
Comment (chengyouling) [reply handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java]: fixed

File: handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java (hunk 2 above)
Comment (chengyouling) [reply handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/ProviderAuthFilter.java]: fixed

**Review by liubao68 — APPROVED** [review]

**Pull request merged**